File: Python/0209.py
from functools import reduce

class Solution:
    def singleNumber(self, nums):
        return reduce(lambda x, y: x ^ y, nums)

# Example usage:
solution = Solution()
print(solution.singleNumber([2, 2, 1]))       # 1
print(solution.singleNumber([4, 1, 2, 1, 2])) # 4
print(solution.singleNumber([1]))             # 1

File: JavaScript/0209.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function(nums) {
    return nums.reduce((acc, num) => acc ^ num, 0);
  };
  
  // Example usage:
  console.log(singleNumber([2, 2, 1]));     // 1
  console.log(singleNumber([4, 1, 2, 1, 2])); // 4
  console.log(singleNumber([1]));            // 1